Update _index.md (#23105)
DataDog/documentation: content/en/developers/integrations/_index.md

Technology Partners write and host the implementation code that makes up the integration. [Creating an API integration][1] works well for Technology Partners building a connector between Datadog and another SaaS platform.

## Requirements
All integrations must include the following:

> * Out-of-the-box Integration Dashboard

Reviewer: ```suggestion
* An Out-of-the-box Integration Dashboard
```

* 3 or more images for your tile
* OAuth - for API Integrations only

> * Log pipeline - If your integration is sending in logs

Reviewer: ```suggestion
* A log pipeline - If your integration sends logs to Datadog
```

> * A log pipeline - If your integration sends logs to Datadog

Author: Should we link out to the associated docs or would this be too much?

Reviewer 2: Since these are in the markdown file and won't be hyperlinks, im not sure they'd get too many clicks if they are buried in here. just my 2c

Author: Agreed - we will keep as is
